### PR TITLE
Add a new option for diagnostic_ignore_dirs

### PR DESCRIPTION
**Pull request by rorydriscoll**
I added a new option to allow people to set directories from which to ignore all diagnostic errors. This has been useful for dealing with third party libraries. I updated the code to match your latest version of SublimeClang, and I fixed a problem with dealing with case insensitivity on Windows.

### CODE REVIEW

File: sublimeclang.py [hunk 1]
```diff
@@ -532,8 +539,10 @@ def display_compilation_results(view):
         return
     errorCount = 0
     warningCount = 0
+    ignoreDirs = [os.path.abspath(os.path.normpath(os.path.normcase(d))) for d in get_setting("diagnostic_ignore_dirs", [], view)]
     try:
         if len(tu.var.diagnostics):
+            print len(tu.var.diagnostics)
```
Comment (quarnster) [inline sublimeclang.py]: @rorydriscoll, looks good save for this print. Could you remove it please?

Comment (rorydriscoll) [inline sublimeclang.py]: Oops. Thanks. I'll do that.

Comment (rorydriscoll) [inline sublimeclang.py]: That's fixed now. Thanks for pointing it out.

**Comment by quarnster** [comment]
Thanks!

**Pull request merged**